src/cmd-buildextend-installer

```diff
@@ -182,19 +202,33 @@ def generate_iso():
         os.chmod(os.path.join(tmpisoimages, file), 0o644)
 
     # Generate initramfs stamp file which says whether it's a live or legacy
-    # initramfs.
+    # initramfs.  Store the build ID in it.
     if is_live:
         stampname = 'etc/coreos-live-initramfs'
     else:
         stampname = 'etc/coreos-legacy-installer-initramfs'
     stamppath = os.path.join(tmpinitrd_base, stampname)
     os.makedirs(os.path.dirname(stamppath), exist_ok=True)
-    open(stamppath, 'w').close()
+    with open(stamppath, 'w') as fh:
+        fh.write(args.build + '\n')
+
+    # Generate rootfs stamp file with the build ID, indicating that the
+    # rootfs has been appended and confirming that initramfs and rootfs are
+    # from the same build.
+    if is_live:
+        stamppath = os.path.join(tmpinitrd_pxe_rootfs, 'etc/coreos-live-rootfs')
+        os.makedirs(os.path.dirname(stamppath), exist_ok=True)
+        with open(stamppath, 'w') as fh:
+            fh.write(args.build + '\n')
 
     # Add osmet files
+    if args.legacy_pxe:
+        tmpinitrd_pxe_or_rootfs = tmpinitrd_pxe
+    else:
+        tmpinitrd_pxe_or_rootfs = tmpinitrd_pxe_rootfs
```

Comment:
Could move this down lower to where it's actually used.

Reply:
Eh, I think it's a bit cleaner to keep it outside the conditionals.